src/osdc/Objecter.cc

```diff
@@ -2316,6 +2319,52 @@ void Objecter::_op_submit(Op *op, shunique_lock<ceph::shared_mutex>& sul, ceph_t
   bool check_for_latest_map = _calc_target(&op->target, nullptr)
     == RECALC_OP_TARGET_POOL_DNE;
 
+  if (op->tid == 0)
+    op->tid = ++last_tid;
+  if (objecter_fastfail_timeout  > timespan(0)) {
+    unique_lock fl(ffpgs.lock);
+    if (ffpgs.pgs.find(op->target.pgid) != ffpgs.pgs.end()) {
+      fl.unlock();
+      ldout(cct, 10) << "fastfail pg: " << op->target.pgid << " to list" << dendl;
+      // before cancelling check if the pg is still inactive
+      // TODO: find a better way to determine if a pg is inactive
+      if(op->target.acting.size() < op->target.min_size) {
+        boost::asio::dispatch(service,
+				      [this, tid=op->tid]() {
+                ldout(cct, 10) << "Cancel op" << dendl;
+                op_cancel(tid, -ETIMEDOUT);
+              });
+      } else {
+        unique_lock fl(ffpgs.lock);
+        ffpgs.pgs.erase(op->target.pgid);
+        fl.unlock();
+      }
+    } else {
+      fl.unlock();
+      op->onfastfail = timer.add_event(objecter_fastfail_timeout,
+				    [this, tid=op->tid, pgid=op->target.pgid, target=op->target, timeout=objecter_fastfail_timeout]() {
+              ldout(cct, 10) << "Enter pg processing" << dendl;
+              unique_lock fl(ffpgs.lock);
+              // TODO: find a better way to determine if a pg is inactive
+				      if (ffpgs.pgs.find(pgid) == ffpgs.pgs.end() && target.acting.size() < target.min_size) {
```

Comment:
what's going on with spacing?

Reply:
I would add parens around both the && clauses, and split 2nd onto the next line--easier to read